feat(apigateway): import existing VpcLink (#7811)
aws/aws-cdk: packages/@aws-cdk/aws-apigateway/README.md

https://{rest-api-id}-{vpce-id}.execute-api.{region}.amazonaws.com/{stage}
```

> ### Importing VPC Link

Reviewer: We should change this to a section called 'VPC Link' and briefly describe with a code snippet about VPC links.

Sorry about the digression since it's not directly related to your change. Are you able to help out? 
Don't forget to add an entry in the 'table of contents' at the top of the file.

A single sentence at the end of that that existing VPC links can be imported using the `fromVpcLinkId()` API should suffice.

Author: @nija-at Yes, of course. I add section about 'VPC Link' and move this in the  'VPC Link' section.

https://{rest-api-id}-{vpce-id}.execute-api.{region}.amazonaws.com/{stage}
```

### VPC Link

> A VPC Link enables access to a resource in your VPC through Amazon API Gateway.
> You create a VPC Link resource targete for one or more network load balancers of the VPC,
> and integrate an API method with a private integration that uses the VpcLink.
> The private integration has an integration type of `HTTP` or `HTTP_PROXY` and has a connection type of `VPC_LINK`.

Reviewer: ```suggestion
A private integration makes it simple to expose HTTP/HTTPS resources behind an
Amazon VPC for access by clients outside of the VPC. The private integration uses
an API Gateway resource of `VpcLink` to encapsulate connections between API
Gateway and targeted VPC resources.

The `VpcLink` is then attached to the `Integration` of a specific API Gateway
Method. The following code sets up a private integration with a network load
balancer -
```

```ts
const vpc = new ec2.Vpc(stack, 'VPC');
const nlb = new elbv2.NetworkLoadBalancer(stack, 'NLB', {
  vpc,
});
const link = new apigw.VpcLink(stack, 'link', {
  targets: [nlb],
});

const integration = new apigw.Integration({
  type: apigw.IntegrationType.HTTP_PROXY,
  options: {
    connectionType: apigw.ConnectionType.VPC_LINK,
    vpcLink: link,
  },
});
```

> You can also import with physical ID of the existing VpcLink resource via the `VpcLink.fromVpcLinkId()`.

Reviewer: ```suggestion
Any existing `VpcLink` resource can be imported into the CDK app via the `VpcLink.fromVpcLinkId()`.
```
